route/src/route_generator_worker.cpp

```diff
@@ -438,5 +442,65 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crossTrackErrorCheck(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
+{
+  if (!world_model_->getMap()) 
+  {
+    ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
+    return true;
+  }
+
+  auto route = world_model_->getRoute()->shortestPath();
+
+  bool out_of_llt_bounds = false;
+
+    lanelet::BasicPoint2d position;
+
+    position.x()= msg->pose.position.x;
+    position.y()= msg->pose.position.y;  
+
+    if (boost::geometry::distance(position, current.polygon2d()) > cross_track_dist)
+        {
+            cte_count++;
+
+            if(cte_count > 4)
+                return true;
+        }
+
+  if(boost::geometry::within(position, current.polygon2d()) == false) //Determine whether or not the vehicle is in the lanelet polygon
+      {
+        out_of_llt_bounds = true;
+      }
+
+    auto following_llts = world_model_->getMapRoutingGraph()->following(current);
+    lanelet::ConstLanelets filtered_llts;//TODO: Filter lanelets from following lanelets that are not a part ot the route
+    int out_count = 0;
+    for(auto i:following_llts)
+    {
+       
+        if (boost::geometry::within(position, i.polygon2d()) == false) 
```

Comment:
I think this would all return false because the current position is in current lanelet, and guaranteed not to be inside following lanelets. I think we just need to find the lanelets on the route by comparing ids.